spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/DefaultDatastoreEntityConverter.java

```diff
@@ -147,8 +147,9 @@ private DatastorePersistentEntity getDiscriminationPersistentEntity(DatastorePer
 
 	private boolean isDiscriminationFieldMatch(DatastorePersistentEntity entity,
 			EntityPropertyValueProvider propertyValueProvider) {
-		return propertyValueProvider.getPropertyValue(entity.getDiscriminationFieldName(), EmbeddedType.NOT_EMBEDDED,
-				ClassTypeInformation.from(String.class)).equals(entity.getDiscriminationValue());
+		return ((String[]) propertyValueProvider.getPropertyValue(entity.getDiscriminationFieldName(),
```

Comment:
should we check if it's null? what happens if the column is not there?

Reply:
This function is only called on persistent entities from the discrimination family, and class types only get added to that family map/tree if they have the appropriate discrimination annotations. Because of that we should only be here if the entity has the column and has the values in the column (or it would have triggered errors elsewhere and earlier)